tracer/test/Datadog.Trace.TestHelpers.AutoInstrumentation/TestHelper.cs

```diff
@@ -673,6 +627,67 @@ protected void EnableTelemetry(bool enabled = true, int? standaloneAgentPort = n
         private bool IsServerSpan(MockSpan span) =>
             span.Tags.GetValueOrDefault(Tags.SpanKind) == SpanKinds.Server;
 
+        private async Task<string> DownloadProcdumpZipAndExtract()
+        {
+            // We don't know if procdump is available, so download it fresh
+            const string url = @"https://download.sysinternals.com/files/Procdump.zip";
+            var client = new HttpClient();
+            var zipFilePath = Path.GetTempFileName();
+            Output.WriteLine($"Downloading Procdump to '{zipFilePath}'");
+            using (var response = await client.GetAsync(url, HttpCompletionOption.ResponseHeadersRead))
+            {
+                using var bodyStream = await response.Content.ReadAsStreamAsync();
+                using Stream streamToWriteTo = File.Open(zipFilePath, FileMode.Create);
+                await bodyStream.CopyToAsync(streamToWriteTo);
+            }
+
+            var unpackedDirectory = Path.Combine(Path.GetTempPath(), Path.GetFileNameWithoutExtension(Path.GetTempFileName()));
+            Output.WriteLine($"Procdump downloaded. Unpacking to '{unpackedDirectory}'");
+            System.IO.Compression.ZipFile.ExtractToDirectory(zipFilePath, unpackedDirectory);
+
+            var procDump = Path.Combine(unpackedDirectory, "procdump.exe");
+            return procDump;
+        }
+
+        private bool CaptureMemoryDump(string tool, string args)
+        {
+            Output.WriteLine($"Capturing memory dump using '{tool} {args}'");
+
+            using var dumpToolProcess = Process.Start(new ProcessStartInfo(tool, args)
+            {
+                UseShellExecute = false,
+                CreateNoWindow = true,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+            });
+
+            dumpToolProcess.OutputDataReceived += (sender, args) =>
+            {
+                if (args.Data != null)
+                {
+                    Output.WriteLine($"[dump][stdout] {args.Data}");
+                }
+            };
+            dumpToolProcess.BeginOutputReadLine();
+
+            dumpToolProcess.ErrorDataReceived += (sender, args) =>
+            {
+                if (args.Data != null)
+                {
+                    Output.WriteLine($"[dump][stderr] {args.Data}");
+                }
+            };
+            dumpToolProcess.BeginErrorReadLine();
+
+            if (!dumpToolProcess.HasExited)
+            {
+                dumpToolProcess.WaitForExit(30_000);
+            }
```

Comment:
nit: could use our existing `ProcessHelper` class for this. Not passionate about it 🙂 
```suggestion
            using var helper = new ProcessHelper(dumpToolProcess);
            dumpToolProcess.WaitForExit(30_000);
            helper.Drain();
            Output.WriteLine($"[dump][stdout] {helper.StandardOutput}");
            Output.WriteLine($"[dump][stderr] {helper.ErrorOutput}");
```